### ISSUE TITLE
unused-argument false-positive with overshadowed variable in dictionary comprehension

### ISSUE THREAD

**Issue by bokhi** [PyCQA/pylint#1731]
### Steps to reproduce
```python
def metadata_from_dict(key):
    return {key: str(value) for key, value in key.items()}
```

### Current behavior
```
W:  1,23: Unused argument 'key' (unused-argument)
```

### Expected behavior
should not raise unused-argument

### pylint --version output
pylint 1.7.4, 
astroid 1.5.3
Python 3.6.2 (default, Jul 17 2017, 23:14:31) 
[GCC 5.4.0 20160609]

**Comment by PCManticore** [comment]
Thanks for the issue @bokhi This makes sense.

**Comment by brycepg** [comment]
Looks like @hippo91 fixed this in pylint 1.8.2 ! Closing
